anillo/middlewares/json.py

```diff
@@ -33,17 +33,23 @@ def wrapper(request, *args, **kwargs):
     return wrapper
 
 
-def wrap_json_body(func):
+def wrap_json_body(func=None, *, preserve_raw_body=False):
     """
     A middleware that parses the body of json requests and
     add it to the request under the `body` attribute (replacing
-    the previous value).
+    the previous value). Can preserve the original value in
+    a new attribute `raw_body` if you give preserve_raw_body=True.
     """
 
+    if func is None:
+        return functools.partial(wrap_json_body, preserve_raw_body=preserve_raw_body)
+
     @functools.wraps(func)
     def wrapper(request, *args, **kwargs):
         ctype, pdict = parse_header(request.headers.get('Content-Type', ''))
         if ctype == "application/json":
+            if preserve_raw_body:
```

Comment:
The second approach is already done: the "wrap_json_params" middleware does exactly that. I understand that "wrap_json_body" is for people that usually wants the body JSON decoded, but perhaps occasionally need to look at the original source.

I'm fine with adding always the raw_body, independently of the content type. Is a good idea. I'll do it.